### PR TITLE
Research articles migration

### PR DESCRIPTION
**Pull request by joewhitsitt**
Relates to https://github.com/uiowa/uiowa/issues/5248

# Todo

- [x] Target the correct content type...
- [x] Redirects
- [x] Article type if external or ianow, link directly to source
- [x] Tags
- [x] Title, Body, Teaser, Summary, other basic Article fields.
- [x] Byline to Organization
- [x] Featured image
- [x] Do not display featured image for all migrated
- [x] Contacts mapping - https://github.com/uiowa/uiowa/issues/5248#issuecomment-1341387260

# How to test

```
ddev blt ds --site=research.uiowa.edu && ddev drush @research.local en research_migrate -y && ddev drush @research.local uli /admin/config/sitenow/sitenow-migrate
```

Complete the form following the help text
`ddev drush @research.local ms`
`ddev drush @research.local mim --all`

Check any migration errors and that the redirects are working. Review the migration deliverables here https://github.com/uiowa/uiowa/issues/5248, particularly https://github.com/uiowa/uiowa/issues/5248#issuecomment-1341387260

Note: there does seem to be a https://research.uiowa.ddev.site/contact-us/cheryl-ridgeway but no migration content is linked to this. Notes say to delete. Could just delete this one manually.

### CODE REVIEW

File: docroot/sites/research.uiowa.edu/modules/research_migrate/src/Plugin/migrate/source/Articles.php
Comment (cory-skeers) [inline docroot/sites/research.uiowa.edu/modules/research_migrate/src/Plugin/migrate/source/Articles.php]: I think if there's a person listed as a contact on the source, but not included in the mapping provided to us, they should be skipped.

@briand44 The issue says that we are not creating people as part of this, but was it specifically so that we don't add new people or because that was an extra complication that wasn't important?

**Comment by cory-skeers** [comment]
I've done a code review and things look good with one question re: creating contact references.

I haven't been able to successfully run the migration locally, though I'm unsure if it's anything here or my environment. My Docker keeps crashing around the 370 - 400 article count. It's not always during the same article, though, so I suspect it's something on my end and unrelated, but want to get someone else testing it or figure this out.

Edit: Still good to check and make sure, but it looks like I might be having some resource limitations in my local environment that's leading to the errors when testing.

File: docroot/sites/research.uiowa.edu/modules/research_migrate/src/Plugin/migrate/source/Articles.php
Comment (briand44) [reply docroot/sites/research.uiowa.edu/modules/research_migrate/src/Plugin/migrate/source/Articles.php]: My understanding was that they have already created all of the contacts as people so there shouldn't be a need to create a new person. If we happen to run into a case where a person doesn't exist, I would be fine with skipping because we don't know if that person should still be the contact or not (possibly left the University)

Comment (cory-skeers) [reply docroot/sites/research.uiowa.edu/modules/research_migrate/src/Plugin/migrate/source/Articles.php]:
```suggestion

```

Comment (joewhitsitt) [reply docroot/sites/research.uiowa.edu/modules/research_migrate/src/Plugin/migrate/source/Articles.php]: That makes sense. Missed that detail.

**Review by cory-skeers — APPROVED** [review]
I was able to get the migration to run to completion and verify it was an issue with my local environment. Went through and spot-checked additional articles after this in addition to the previous code review. Think this looks good to go!

**Comment by joewhitsitt** [comment]
Thanks for reviewing @cory-skeers. What was the issue/solution locally?

@briand44 should we run this in DEV for customer review before merging?

**Comment by briand44** [comment]
Where do we go from here? Do we sync the .prod site down to dev and make dev available and run this in dev for review?

**Comment by briand44** [comment]
>should we run this in DEV for customer review before merging?

Yes, that would be great

**Comment by cory-skeers** [comment]
> Thanks for reviewing @cory-skeers. What was the issue/solution locally?

Temporarily, just cranking up the allowed disk space for docker. I think there's still and underlying issue, since there's weird stuff like it claiming to be using 0B of space, but at least this keeps it running.

**Comment by briand44** [comment]
It looks like there are a handful of articles that already exist on the .prod site. We may need to delete those pre/post migration in dev/prod if they end up being duplicates.

**Comment by joewhitsitt** [comment]
> It looks like there are a handful of articles that already exist on the .prod site. We may need to delete those pre/post migration in dev/prod if they end up being duplicates.

I'll dump them for DEV, but it would be good to get clarity on that for PROD.

**Comment by joewhitsitt** [comment]
Holding off on syncing research.dev a fresh db copy until we can confirm they aren't using it currently.

**Comment by briand44** [comment]
>Holding off on syncing research.dev a fresh db copy until we can confirm they aren't using it currently.

Modei gave the all clear. We can move forward, they are not using dev.

**Comment by joewhitsitt** [comment]
It appears to have choked running mim --all. Setting things up again to run tomorrow AM.

```
322/412 [=====================>------]  78% [warning] Memory usage is 439.23 MB (86% of limit 512 MB), reclaiming memory.
 [warning] Memory usage is now 412.64 MB (81% of limit 512 MB), not enough reclaimed, starting new batch
 [notice] Processed 322 items (322 created, 0 updated, 0 failed, 0 ignored) - done with 'research_articles'
 [error]  Migration research_articles_redirects did not meet the requirements. Missing migrations research_articles. requirements: research_articles.
```

**Comment by joewhitsitt** [comment]
Went better with `--limit`
https://research.dev.drupal.uiowa.edu/news

```
ddev drush @research.dev mim research_articles --limit=100
 [notice] Processed 100 items (100 created, 0 updated, 0 failed, 0 ignored) - done with 'research_articles'
Connection to uiowadev.ssh.prod.acquia-sites.com closed.
jwhitsit@ITS-02342 uiowa % ddev drush @research.dev mim research_articles --limit=100
   1/412 [>---------------------------]   0% [error]  Invalid marker found at offset 0: 0x47 
  76/412 [=====>----------------------]  18% [error]  File at 'public://2022-06/2017_06_26-kerbeshian-marie-v3.jpg' does not exist (plugin: imagemagick_identify) (method: loadMetadataFromFile) 
 [error]  File at 'public://2022-06/2017_06_26-kerbeshian-marie-v3.jpg' does not exist (plugin: imagemagick_identify) (method: loadMetadataFromFile) 
 [notice] Processed 100 items (100 created, 0 updated, 0 failed, 0 ignored) - done with 'research_articles'
Connection to uiowadev.ssh.prod.acquia-sites.com closed.
jwhitsit@ITS-02342 uiowa % ddev drush @research.dev mim research_articles --limit=100
  52/412 [===>------------------------]  12% [error]  Unknown format: 0x300 
 [notice] Processed 100 items (100 created, 0 updated, 0 failed, 0 ignored) - done with 'research_articles'
Connection to uiowadev.ssh.prod.acquia-sites.com closed.
jwhitsit@ITS-02342 uiowa % ddev drush @research.dev mim research_articles --limit=100
  84/412 [=====>----------------------]  20% [error]  Unknown format: 0x300 
  95/412 [======>---------------------]  23% [error]  Invalid marker found at offset 0: 0x89 
 [notice] Processed 100 items (100 created, 0 updated, 0 failed, 0 ignored) - done with 'research_articles'
Connection to uiowadev.ssh.prod.acquia-sites.com closed.
jwhitsit@ITS-02342 uiowa % ddev drush @research.dev mim research_articles --limit=100
 [notice] Possible broken links found in node 806: here: node&quot; data-entity-uuid=&quot;9d406e7e-0c06-4943-b3c9-0ef7bfefd22d&quot; href=&quot;/node/581
 [notice] Possible broken links found in node 856: Visit our website: node/270
 [notice] Possible broken links found in node 1421: Internal Funding Initiatives: node/270,.: node/289
 [notice] Possible broken links found in node 1736: Arts &amp;amp; Humanities Initiative (AHI) program: node/270,Visit our website: node/270
 [notice] Possible broken links found in node 1791: Creative Matches grant program: node/2591,Visit our website: node/2591
 [notice] Possible broken links found in node 1886: John E. Sawyer Seminars program: node/546,OVPRED&amp;nbsp;website for the Sawyer Seminars: node/546
 [notice] Possible broken links found in node 1906: Visit our website: node/2591
 [notice] Possible broken links found in node 1926: &lt;span style=&quot;color:#954f72&quot;&gt;http://www.iowabio.org/index.cfm?NodeID=91057&lt;/span&gt;: NodeID=91057
 [notice] Possible broken links found in node 1956: Humanities Advisory Board: node/262
 [notice] Possible broken links found in node 2001: mental health and illness: node/3581
 [notice] Possible broken links found in node 2016: Visit our website: node/289
 [notice] Possible broken links found in node 2101: Research Services Fair: node/4116
 [notice] Possible broken links found in node 2341: common IT services: node/118001?utm_source=working-remotely-FAQ-page&amp;amp;utm_medium=coronavirus-website&amp;amp;utm_campaign=working-remotely
 [notice] Possible broken links found in node 2361: &lt;span style=&quot;background-color:white&quot;&gt;&lt;span style=&quot;background-position:initial initial&quot;&gt;&lt;span style=&quot;background-repeat:initial initial&quot;&gt;remote access&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;: node%2F118001%3Futm_source%3Dworking-remotely-FAQ-page%26utm_medium%3Dcoronavirus-website%26utm_campaign%3Dworking-remotely&amp;amp;e=se&amp;amp;p=web&amp;amp;tv=no-js-0.1.0&amp;amp;aid=edu.uiowa.its.ais.dispatch&amp;amp;se_ca=email&amp;amp;se_ac=link-click&amp;amp;se_la=remote+access&amp;amp;se_pr=51796fd7-fbab-4b33-af3b-deed1fc4bc50&amp;amp;cx=eyJzY2hlbWEiOiJpZ2x1OmNvbS5zbm93cGxvd2FuYWx5dGljcy5zbm93cGxvdy9jb250ZXh0cy9qc29uc2NoZW1hLzEtMC0wIiwiZGF0YSI6W3sic2NoZW1hIjoiaWdsdTplZHUudWlvd2EuaXRzL2Rpc3BhdGNoX21lc3NhZ2UvanNvbnNjaGVtYS8xLTAtMCIsImRhdGEiOnsic3ViamVjdCI6IlRFU1QgLSBbVW5pdkFkbV0gQ09WSUQtMTkgTmVjZXNzaXRhdGVzIFJlc2VhcmNoIEFjdGl2aXR5IFJhbXAtZG93biIsImZyb21BZGRyZXNzIjoibWFydHktc2Nob2x0ekB1aW93YS5lZHUiLCJiYXRjaElkIjoiNzU4NDYzNzg0Iiwic2VudCI6IjAzLzE4LzIwMjAgMTQ6NDQ6NDYiLCJtZW1iZXJJZCI6IjUxNzk2ZmQ3LWZiYWItNGIzMy1hZjNiLWRlZWQxZmM0YmM1MCJ9fV19&amp;amp;uid=89d246a869d3b9aeefaac5b98377fc08af6de96f&quot; style=&quot;color:blue; text-decoration:underline&quot; target=&quot;_blank
 [notice] Possible broken links found in node 2426: SHL Helps Test Iowa Reach 100,000 Milestone: node/5851
 [notice] Possible broken links found in node 2631: initiative’s webpage: node/6441
 [notice] Possible broken links found in node 2651: AHI webpage: node/270
 [notice] Possible broken links found in node 2736: solicits nominations from the campus community: node/7201
 [notice] Possible broken links found in node 2746: Seeding Excellence Initiative: node/6976
 [notice] Possible broken links found in node 2751: UI State Hygienic Lab plays critical role in monitoring air quality for Iowans: node/6906
 [notice] Possible broken links found in node 2771: Write Winning Grant Proposals seminar: node/264,Register here. : node/264
 [notice] Possible broken links found in node 2871: CLAS professors receive funding for community engaged scholarship: node/7896
 [notice] Processed 12 items (12 created, 0 updated, 0 failed, 0 ignored) - done with 'research_articles'
Connection to uiowadev.ssh.prod.acquia-sites.com closed.
jwhitsit@ITS-02342 uiowa % ddev drush @research.dev mim research_articles_redirects 
 [notice] Processed 412 items (412 created, 0 updated, 0 failed, 0 ignored) - done with 'research_articles_redirects'
Connection to uiowadev.ssh.prod.acquia-sites.com closed.
```

**Comment by pyrello** [comment]
What is the status of this PR?

**Comment by joewhitsitt** [comment]
Customer is reviewing results of migration in DEV

**Comment by joewhitsitt** [comment]
> We got approval to move ahead and run the migration in prod. - https://github.com/uiowa/uiowa/issues/5248#issuecomment-1372867336

With no changes since technical approval. Getting ready for merge.

**Pull request merged**
